test/functional/p2p_compactblocks_blocksonly.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that a node in blocksonly mode does not request compact blocks."""
+
+from test_framework.messages import (
+    MSG_BLOCK,
+    MSG_CMPCT_BLOCK,
+    MSG_WITNESS_FLAG,
+    CBlock,
+    CBlockHeader,
+    CInv,
+    from_hex,
+    msg_block,
+    msg_getdata,
+    msg_headers,
+    msg_sendcmpct,
+)
+from test_framework.p2p import P2PInterface
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class P2PCompactBlocksBlocksOnly(BitcoinTestFramework):
+    def set_test_params(self):
+        self.extra_args = [["-blocksonly"], [], [], []]
+        self.num_nodes = len(self.extra_args)
```

Comment:
I think this would be clearer if it was explicit about the number of nodes:

```suggestion
        self.num_nodes = 3
```

Checking the number of test nodes is one of the first things I do when reading a functional test. It should be immediately obvious what the topology of the test is.